src/relay/src/readInlineData.js

```diff
@@ -4,9 +4,12 @@ import { readInlineData as _readInlineData } from 'react-relay';
 
 import type { GraphQLTaggedNode } from './types.flow';
 
-export default function readInlineData(
-  fragment: GraphQLTaggedNode,
-  fragmentRef: GraphQLTaggedNode,
-) {
+opaque type FragmentReference = empty;
+
+export default function readInlineData<
+  TRef: FragmentReference,
+  TData,
+  TKey: ?{ +$data?: TData, +$fragmentRefs: TRef, ... },
+>(fragment: GraphQLTaggedNode, fragmentRef: TKey): ?TData {
```

Comment:
Seems like `TData` should be nullable only when `fragmentRef` is nullable, see: https://github.com/facebook/relay/blob/bb33efe09529e231ecccdbaaf96cb0fd69f86301/packages/relay-runtime/store/readInlineData.js#L25-L40

I will merge it so we can iterate.

Reply:
Yeah, I chose to go with the latter, ok, let's iterate on it.

Reply:
@tbergq It would be better to go with both. It's like a variant. Similarly for example `JSON.stringify` [is also defined twice](https://github.com/facebook/flow/blob/f2a168ebbc71c6bc51c8565768b11c6432a8d216/lib/core.js#L511-L520) since it's more complicated in reality.